modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/SourceModelLoaderDelegate.java

```diff
@@ -104,10 +108,35 @@ void declareMessageSources(ExtensionDeclarer extensionDeclarer,
       declareSourceCallbackParameters(parser.getOnTerminateCallbackParser(), sourceDeclarer::onTerminate);
       declareSourceCallbackParameters(parser.getOnBackPressureCallbackParser(), sourceDeclarer::onBackPressure);
 
+      declareClusterSupport(sourceDeclarer, parser.getSourceClusterSupport());
+
+      sourceDeclarer.withModelProperty(parser.getBackPressureStrategyModelProperty().orElse(getDefault()));
+
       sourceDeclarers.put(parser, sourceDeclarer);
     }
   }
 
+  private void declareClusterSupport(SourceDeclarer sourceDeclarer, Optional<SourceClusterSupport> sourceClusterSupport) {
+    boolean runsOnPrimaryNodeOnly;
+    SourceClusterSupport resultingSourceClusterSupport;
+    switch (sourceClusterSupport.orElse(DEFAULT_ALL_NODES)) {
```

Comment:
this switch doesn't belong there. The loader should just set sourceClusterSupport on the declarer. The logic about how that affects the `runsOnPrimaryNodeOnly` is a language level concern.

Reply:
I would take the configuration of the `runsOnPrimaryNodeOnly` to the enricher right? 

Regarding the switch, I would still need it to get the adequate SourceClusterSupport for the model property.

Reply:
right. The point is that setting the  `runsOnPrimayNodeOnly` property in the `SourceModel` is not something that should be left up to the loader since many things depend on that. To clarify, the problem is not the switch itself, but its location